phpstan.neon.dist

```diff
@@ -1,9 +1,28 @@
 includes:
     - ./vendor/larastan/larastan/extension.neon
 parameters:
-    level: 1
+    level: 2
     paths:
         - src
         - tests
     ignoreErrors:
         - '#Unsafe usage of new static\(\).#'
+        - '#Access to an undefined property Staudenmeir\\LaravelAdjacencyList\\Tests\\Tree\\Models\\[a-zA-Z\\]+::\$[a-zA-Z0-9_]+#'
+        - '#Access to an undefined property Illuminate\\Database\\Query\\Builder::\$expressions#'
+        - '#Access to an undefined property Illuminate\\Database\\Eloquent\\Model::\$laravel_through_key#'
```

Comment:
I think these are properties on other builders/models which are not properly typed ATM

Reply:
@SanderMuller I resolved the first error with type hints.

Do you know a way to resolve the second one with type hints? Something like an inline `@var` for properties?

Reply:
@staudenmeir nice work so far!

I've worked towards this via 5ae8a0528f1bf93473eb1cf3801fc3d95ecb9788, please let me know what you think about this direction. There are now just 7 errors remaining, I think all of those are related to the `Staudenmeir\EloquentHasManyDeep\HasOneDeep` relation getting resolved as a collection of results instead of a single result. I think to resolve this, `Staudenmeir\EloquentHasManyDeep\HasOneDeep` should extend another class than `HasManyDeep`, perhaps `HasOneThrough`, although this would require similar code to be added as in `HasManyDeep` which could be done via traits instead, I suppose